saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/node/Node.scala

```diff
@@ -79,23 +77,23 @@ class Node[T <: AnyRef](
     node
   }
 
-  def clear() = {
-    collection.clear()
-    trainingSet.clear()
-    testingSet.clear()
+  def clear(): Unit = {
+    collection.clear
+    trainingSet.clear
+    testingSet.clear
     for (e <- incoming) e.clear
```

Comment:
I saw above  in `nodes.foreach(_.clear())` you added parentheses and here you removed them, since you are cleaning up maybe stay consistent.